### PR TITLE
feat: Validate responses and log any errors that occur + make index file append more atomic

### PR DESCRIPTION
**Pull request by lukehesluke**
## QA

SessionSeries feed DL time: 2 mins
ScheduledSession feed DL time: 3 mins

### Error cases

For these error cases, I temporarily changed max exponential backoff to 4s

URL doesn't exist:

Screenshot 2020-08-19 at 11 31 20

404:

Screenshot 2020-08-19 at 12 32 24

JSON but not valid:

Screenshot 2020-08-19 at 11 34 46

### Success cases

After DLing SeS feed, index file:

Screenshot 2020-08-19 at 11 44 20

the 1st data file:

Screenshot 2020-08-19 at 11 45 29

Scheduled Session index file:

Screenshot 2020-08-19 at 12 05 45

A ScS data file:

Screenshot 2020-08-19 at 12 06 55

### Do SeS and ScS match?

I couldn't ask this question too many times because the sheer load of ScheduledSessions in files make this a very time-costly query. But here's one example of a match:

Screenshot 2020-08-19 at 12 10 53

### CODE REVIEW

File: walkRpde.js [hunk 1]
```diff
@@ -7,12 +7,23 @@ const { default: axios } = require('axios');
...
 
 /**
  * Set this to true in order to log timings of tasks within the script
  */
 const DO_DEBUG_PERFORMANCE = false;
 
+// Parameters controlling retry backoff
+const RETRY_BACKOFF_MIN = 1;
+// const RETRY_BACKOFF_MAX = 1024; // ~ 17 mins
+const RETRY_BACKOFF_MAX = 4;
```
Comment (nickevansuk) [inline walkRpde.js]: Should this not be left as 1024?

**Review by nickevansuk — CHANGES_REQUESTED** [review]
(from slack)
Should it abort on 404, same as test suite? (e.g. if the firehose URL is wrong)?

Otherwise one minor thing=

File: walkRpde.js (hunk 1 above)
Comment (lukehesluke) [reply walkRpde.js]: Yep. Fixed

**Review by nickevansuk — APPROVED** [review]
Looks good!

**Pull request merged**